drivers/xen/dom0/domctl.c

```diff
@@ -0,0 +1,302 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (c) 2023 EPAM Systems
+ *
+ */
+
+#include <zephyr/arch/arm64/hypercall.h>
+#include <zephyr/xen/dom0/domctl.h>
+#include <zephyr/xen/generic.h>
+#include <zephyr/xen/public/domctl.h>
+#include <zephyr/xen/public/xen.h>
+
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <string.h>
+
+static int do_domctl(xen_domctl_t *domctl)
+{
+	domctl->interface_version = XEN_DOMCTL_INTERFACE_VERSION;
+	return HYPERVISOR_domctl(domctl);
+}
+
+int xen_domctl_scheduler_op(int domid, struct xen_domctl_scheduler_op *sched_op)
+{
+	xen_domctl_t domctl = {0};
```

Comment:
(note) you can use the next initializer here and below:
```
xen_domctl_t domctl = {
	.cmd = XEN_DOMCTL_scheduler_op,
	.domain = domid,
	.u.scheduler_op = *sched_op,
};
```
and rest of fields automatically will be initialized

Reply:
@xakep-amatop thanks for suggestion, done.